src/app/components/AmountFormatter/index.tsx

```diff
@@ -23,31 +24,43 @@ export interface AmountFormatterProps {
 /**
  * Formats base unit amounts to ROSEs
  */
-export const AmountFormatter = memo((props: AmountFormatterProps) => {
-  const ticker = useSelector(selectTicker)
-  if (props.amount == null) return <>-</>
+export const AmountFormatter = memo(
+  ({
+    amount,
+    amountUnit = 'baseUnits',
+    minimumFractionDigits,
+    maximumFractionDigits,
+    hideTicker,
+    size,
+    smallTicker,
+  }: AmountFormatterProps) => {
+    const ticker = useSelector(selectTicker)
+    const isUsingBaseUnits = amountUnit === 'baseUnits'
+    if (amount == null) return <>-</>
 
-  const amountString = formatBaseUnitsAsRose(props.amount, {
-    minimumFractionDigits: props.minimumFractionDigits ?? 1,
-    maximumFractionDigits: props.maximumFractionDigits ?? 15,
-  })
+    const formatter = isUsingBaseUnits ? formatBaseUnitsAsRose : formatWeiAsWrose
+    const amountString = formatter(amount, {
+      minimumFractionDigits: minimumFractionDigits ?? 1,
+      maximumFractionDigits: maximumFractionDigits ? maximumFractionDigits : isUsingBaseUnits ? 15 : 18,
```

Comment:
I don't think this works well either, we use maximumFractionDigits={0} in some places :D